src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -439,21 +436,14 @@ private static string GetProjectIdentifier(string projectKey, string projectBran
             return projectId;
         }
 
-        private string GetUrl(string format, params string[] args)
-        {
-            var queryString = EscapeQuery(format, args);
-            if (!queryString.StartsWith("/", StringComparison.OrdinalIgnoreCase))
-            {
-                queryString = string.Concat('/', queryString);
-            }
+        private Uri GetUri(string query, params string[] args) =>
+            new(serverUrl, Escape(query, args));
 
-            return this.serverUrl + queryString;
-        }
+        private static string Escape(string format, params string[] args) =>
+            string.Format(CultureInfo.InvariantCulture, format, args.Select(Escape).ToArray());
```

Comment:
I find this easier to understand, without the additional rename
```suggestion
            string.Format(CultureInfo.InvariantCulture, format, args.Select(WebUtility.UrlEncode).ToArray());
```